### PR TITLE
Birmingham | ITP-Jan | Ahmad Roman Sanaye | Sprint 3 | Practice-tdd

### PR DESCRIPTION
**Pull request by RomanSanaye**
Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the style guide

## Changelist

Implemented the following functions:

- count
- getOrdinalNumber
- repeatStr

Each function handles the required cases and passes all Jest test cases, including edge cases (positive, zero, and negative values where applicable).

All tests are passing successfully.

### CODE REVIEW

File: Sprint-3/2-practice-tdd/count.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+    return 0;
+  }
+  let numberOfChar = 0;
+  for (let wantedChar of stringOfCharacters) {
+    if (wantedChar === findCharacter) {
+      numberOfChar++;
+    }
+    if (numberOfChar === 0) {
+     return `0 ${findCharacter} found.`
+    }
```
Comment (cjyuan) [inline Sprint-3/2-practice-tdd/count.js]: The function is expected to return an integer number. Returning a string would make the function difficult to use because the caller would need to check the data type of the return value.

File: Sprint-3/2-practice-tdd/get-ordinal-number.test.js [hunk 2]
```diff
@@ -18,3 +18,27 @@ test("should append 'st' for numbers ending with 1, except those ending with 11"
...
+  expect(getOrdinalNumber(2)).toEqual("2nd");
+  expect(getOrdinalNumber(22)).toEqual("22nd");
+  expect(getOrdinalNumber(222)).toEqual("222nd");
+});
+test("should append 'rd' for numbers ending with 3, except those ending with 13", () => {
+  expect(getOrdinalNumber(3)).toEqual("3rd");
+  expect(getOrdinalNumber(33)).toEqual("33rd");
+  expect(getOrdinalNumber(333)).toEqual("333rd");
+});
+test("should append 'th' for numbers ending with 4 and more than 4", () => {
```
Comment (cjyuan) [inline Sprint-3/2-practice-tdd/get-ordinal-number.test.js]: `... ending with 4-9, 0` might be more concise.

File: Sprint-3/2-practice-tdd/repeat-str.test.js [hunk 3]
```diff
@@ -10,23 +10,27 @@ const repeatStr = require("./repeat-str");
...
-  expect(repeatedStr).toEqual("hellohellohello");
+  expect(repeatStr("Love", 3)).toEqual("LoveLoveLove");
 });
 
 // Case: handle count of 1:
 // Given a target string `str` and a `count` equal to 1,
 // When the repeatStr function is called with these inputs,
 // Then it should return the original `str` without repetition.
-
+test("should print the original string", () => {
```
Comment (cjyuan) [inline Sprint-3/2-practice-tdd/repeat-str.test.js]: "print" may suggest the function should output the string.

File: Sprint-3/2-practice-tdd/repeat-str.test.js [hunk 4]
```diff
@@ -10,23 +10,27 @@ const repeatStr = require("./repeat-str");
...
+  expect(repeatStr("Love", 1)).toEqual("Love");
+});
 // Case: Handle count of 0:
 // Given a target string `str` and a `count` equal to 0,
 // When the repeatStr function is called with these inputs,
 // Then it should return an empty string.
-
+test("should return an empty string", () => {
```
Comment (cjyuan) [inline Sprint-3/2-practice-tdd/repeat-str.test.js]: You could probably make this description more informative by including the `when ...` part.

**Review by cjyuan — COMMENTED** [review]
I noticed some minor format inconsistency.

Have you installed the **prettier** VSCode extension and enabled "Format on save/paste" on VSCode, as recommended in 
https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/blob/main/readme.md
?

If you have enabled "Format on save" but it is not working, it is likely that you haven't assign a formatter for JS file. This could happen if you have zero or multiple extensions that can format .js file.

If you have installed "Prettier" extension. To assign it as the formatter of JS code, you can try:
1. Use "Format document" to format the JS file. Sometimes, VSCode will ask you to choose a formatter, and you can manually select "Prettier".
2. Edit `settings.json` and set Prettier as the default formatter for JS.
See: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

**Comment by cjyuan** [comment]
There is no new commits on this branch, and the comments are not yet addressed.

If you have responded to the comments, you may want to check if the responses are in "Pending" state. Other users cannot see the pending comments.

File: Sprint-3/2-practice-tdd/count.js (hunk 1 above)
Comment (RomanSanaye) [reply Sprint-3/2-practice-tdd/count.js]: Hello @cjyuan ,
I have updated the function so it now always returns an integer. Returning 0 instead of a string ensures the function is easier to use and matches the expected return type.

File: Sprint-3/2-practice-tdd/repeat-str.test.js (hunk 3 above)
Comment (RomanSanaye) [reply Sprint-3/2-practice-tdd/repeat-str.test.js]: I have updated the test description to say “return” instead of “print” to accurately reflect that the function returns a string rather than printing it.

File: Sprint-3/2-practice-tdd/repeat-str.test.js (hunk 4 above)
Comment (RomanSanaye) [reply Sprint-3/2-practice-tdd/repeat-str.test.js]: I have revised the test descriptions to include the input condition (when …) so they clearly state under what circumstances the function should return the expected result.

File: Sprint-3/2-practice-tdd/get-ordinal-number.test.js (hunk 2 above)
Comment (RomanSanaye) [reply Sprint-3/2-practice-tdd/get-ordinal-number.test.js]: I have updated the ordinal number test descriptions to group numbers ending in 4-9 and 0, making the tests more concise while still covering all cases.

File: Sprint-3/2-practice-tdd/count.js [hunk 5]
```diff
@@ -1,5 +1,18 @@
...
+  for (let wantedChar of stringOfCharacters) {
+    if (wantedChar === findCharacter) {
+      numberOfChar++;
+    }
+    if (numberOfChar === 0) {
+      return 0;
+    }
+  }
+  return numberOfChar;
 }
```
Comment (cjyuan) [inline Sprint-3/2-practice-tdd/count.js:13]: This function has a bug; it does not always return the correct count.

If you cannot figure out the bug, try testing a few more samples in `count.test.js`.

**Comment by RomanSanaye** [comment]
Hi @cjyuan , thanks for your feedback.

I have addressed all the comments:

* Updated the function to always return a number instead of a string
* Fixed the bug in `countChar` by removing the early return inside the loop so it now correctly counts all characters
* Improved test descriptions to use “return” instead of “print”
* Made test descriptions more informative by including the “when…” part
* Simplified ordinal number tests by grouping cases (e.g. numbers ending in 4–9 and 0)

I have also added the **“Needs Review”** label.

Please let me know if anything else is required.

**Comment by cjyuan** [comment]
Changes look good, and thanks for the clear explanation in the responses. Well done.

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
